Update proposals-CG-WG.md (#12)
w3c-fedid/Administration: proposals-CG-WG.md

Author: @samuelgoto

> Created: 08/11/2024

Reviewer: We're international. [ISO 8601](https://xkcd.com/1179/), please.
```suggestion
Created: 2024-08-11
```

Reviewer: Is this August 11 or November 8?

ISO 8601 is unambiguous, if used consistently.

> Last update: 11/26/2024

Reviewer: As above.
```suggestion
Last update: 2024-11-26
```

* [Stage 3](#stage3): **Implementation** of the preferred Proposal
* [Stage 4](#stage4): **Publication** of a Proposed Recommendation

> These stages support the W3C process and align with the [TC39 process](https://tc39.es/process-document/) and the [WHATWG process](https://whatwg.org/stages). If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedent.

Reviewer 2: ```suggestion
These stages support the W3C process and align with the [TC39 process](https://tc39.es/process-document/) and the [WHATWG process](https://whatwg.org/stages). If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedence.
```

> These stages support the W3C process and align with the [TC39 process](https://tc39.es/process-document/) and the [WHATWG process](https://whatwg.org/stages). If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedent.

Reviewer 3: Can you revert this change too as it is covered in this other PR? https://github.com/w3c-fedid/Administration/pull/13/files#diff-413347616d7d3063a428e05c5c47d90aa1301ba3dade010736161ffb20774cedR16

The purpose of Stage 1 proposals is to explore the Solution Space and pick a preferred direction. This is where most of the work gets done because it involves exploring alternatives, understanding tradeoffs, gathering implementation experience, incubating alternatives, gathering evidence of demand and fitness for purpose, and finally, identifying the best out of the many alternatives.

  * What's needed from individuals?
    * [ ] Identification of [champions](https://github.com/tc39/how-we-work/blob/main/champion.md).
    * [ ] An understanding of the Problem Space.
    * [ ] Optionally, alternatives under consideration.
  * What's asked of the **Community Group**?
    * [ ] Community Group consensus that the problem is worth spending the Community Group’s time working on.
    * [ ] A home for incubating the proposal. Small features will incubate in issues. If and when the champions are ready to more thoroughly document their proposal, the WG chairs will create a repo for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/LightweightFedCM)). 

# Stage 2: Formalization

> The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal [Draft Report](https://www.w3.org/community/about/process/#deliverables).

Reviewer: ```suggestion
The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of issues blocking advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal [Draft Report](https://www.w3.org/community/about/process/#deliverables).
```

The purpose of Stage 1 proposals is to explore the Solution Space and pick a preferred direction. This is where most of the work gets done because it involves exploring alternatives, understanding tradeoffs, gathering implementation experience, incubating alternatives, gathering evidence of demand and fitness for purpose, and finally, identifying the best out of the many alternatives.

  * What's needed from individuals?
    * [ ] Identification of [champions](https://github.com/tc39/how-we-work/blob/main/champion.md).
    * [ ] An understanding of the Problem Space.
    * [ ] Optionally, alternatives under consideration.
  * What's asked of the **Community Group**?
    * [ ] The Community Group consensus that the problem is worth spending the Community Group’s time working on.
    * [ ] A home for incubating the proposal. Small features will incubate in issues. If and when the champions are ready to more thoroughly document their proposal, the WG chairs will create a repo for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/LightweightFedCM)). 

# Stage 2: Formalization

> The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal draft.

Reviewer: ```suggestion
The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of issues that block advancement to the next stage, and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal draft.
```

Reviewer: Existing language is not proper English construction. I suggested changes to make it proper English. Why is this marked resolved without application of those changes?

> The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal draft.

Reviewer 4: Complete formal draft of what? I guess the specification of the feature?

Author: Yes, specification of the feature.

Reviewer 5: W3C CGs use *Draft Report* and *Final Report* terminology. Could **draft report** make it more clear?
https://www.w3.org/community/about/process/#deliverables
![image](https://github.com/user-attachments/assets/4f358a4b-a222-483c-92ee-44667cabd28c)

  * What's needed from champions?

>     * [ ] An [explainer](https://tag.w3.org/explainers/).

Reviewer 3: Can I suggest that we keep these editorial fixes outside of this PR so that we can focus on the normative changes? For example, I almost dismissed the change below because I assumed that it was just making editorial changes rather than normative.

The editorial changes are perfectly valid, I just think that they may be best done in isolation rather than together with this spec PR.

Reviewer 3: can you revert the non-editorial changes so that it is easier to review?

Reviewer: > The editorial changes are perfectly valid, I just think that they may be best done in isolation rather than together with this spec PR.

Often, editorial changes make a significant difference to the comprehensibility of a spec, which makes a difference in reviewing the technical changes. Certainly, editorial changes of the lines impacted by the PR should be allowed, because such in-PR changes are a substantially lighter lift for the reviewer/commenter than a distinct PR — and don't require that the reviewer remember, once the technical PR is merged, that they wanted to make an editorial change.

    * [ ] Documentation of alternatives and trade-offs considered.

>     * [ ] Draft specification text (or detailed examples/code samples if needed for clarity).

Reviewer 3: This seems like a higher bar than is necessary. I'd suggest allowing proposals to enter Stage 2 without a draft specification text (explainers are necessary).

```suggestion
    * [ ] Optionally, a draft specification text (or detailed examples/code samples if needed for clarity).
```

    * [ ] Early implementation experience, including prototypes or trials.

>     * [ ] Evidence of stakeholder demand and use-case alignment.

Reviewer 3: Stakeholders seems unnecessarily underspecified as far as "demand" is concerned. Where else would "demand" come from other than web developers?

```suggestion
    * [ ] Evidence of stakeholder (e.g. web developers) demand and use-case alignment.
```

Reviewer: Minor tweaks to @samuelgoto's suggestion
```suggestion
    * [ ] Evidence of stakeholder (e.g., web developer) demand and use-case alignment.
```

  * What's asked of the **Working Group**?
    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.

>     * [ ] A completed Working Draft for further iteration.

Reviewer 3: What does this mean? What does it mean that we ask the Working Group for a Working Draft?

That is, do we expect every different proposal to have separate working drafts?

Reviewer 4: As a spec editor, I would not enjoy having a separate 'working draft' for every feature. That said, maybe this can say 'pull request' at least for features that are intended to be part of an existing spec and not their own spec (which are probably the vast majority)?

Author: If it's an update, then yes, a PR. If it's a whole new thing, then a doc draft.

>     * [ ] Approval to transition the draft into a formal as the basis for their work as a [Working Draft](https://www.w3.org/policies/process/#RecsWD).

Reviewer 3: I don't follow this: do we expect that, when a proposal enters Stage 2, that it will have the proposal merged into the Working Draft?

What does this mean? What does it mean that we ask the Working Group for a Working Draft?

That is, do we expect every different proposal to have separate working drafts?

Or is it that what one gets is approval to spend time in the WG to review spec PRs?

```suggestion
    * [ ] Approval to dedicate the WG's time to Review spec PRs to merge the proposal into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).
```

  
# Stage 3: Implementation

> NOTE: We don't yet have enough experience as we get towards Stage 3 and 4, so they aren't fully baked yet. It is possible that these
> will change a lot as we learn, including potentially entirely replacing them with references to the W3C Process. 

> The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR). The spec should be considered finished, pending editorial nit review. However, since multiple implementations are expected in this stage, it is possible there will be further changes to normative content.

Reviewer: ```suggestion
The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR). The spec should be considered finished, pending editorial nit review. However, since multiple implementations are expected to be produced during this stage, it is possible there will be further changes to normative content based on feedback from those implementors.
```

 
  * What's needed from champions?
    * [ ] Proposal fully merged into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).
    * [ ] Web Platform Tests are available.
    * [ ] At least two independent implementers and no unresolved objections (where resolution may include an agreement not to address the issue).
  * What's asked of the **Working Group**? 

>     * [ ] Working Group consensus that the [Working Draft](https://www.w3.org/policies/process/#RecsWD) sufficiently resolves all of the issues raised at [Stage 2](#stage-2).

Reviewer 4: Since we are modifying this, maybe we should address the issue about not all features within a spec reaching Stage 3 at the same time. Perhaps the CR mentioned below can be a frozen branch from the WD, possibly with features in lesser stages removed?

Author: I don't think that's necessary, though if you feel strongly about it, we'll have to come up with text for both stage 2 and 3, as neither require all features reach the stage at the same time.

    * [ ] Working Group consensus to publish the [Working Draft](https://www.w3.org/policies/process/#RecsWD) as the Working Group's [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR).
  
# Stage 4: Publication

>  The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C). At this stage, the spec is merged and has finished editor review. This editor review could be lengthy, especially if the feature is large and/or the contributor is new to W3C process, but it will usually be short.

Reviewer: ```suggestion
 The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C). At this stage, the spec is merged and has finished editorial review. This editorial review could be lengthy, especially if the feature is large and/or the contributor is new to W3C process, but it will usually be short.
```

 
  * What's needed from champions?
    * [ ] Documented [implementation experience](https://www.w3.org/policies/process/#implementation-experience) across multiple environments.
  * What's asked of the **Working Group**?
    * [ ] Working Group consensus that the [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR)'s implementation experience is complete.
    * [ ] Working Group consensus to publish it as a [Proposed Recommendation](https://www.w3.org/policies/process/#RecsPR).

# Prior Art

* TC39  
  * Proposals Repo: [https://github.com/tc39/proposals](https://github.com/tc39/proposals)  
  * Stages: [https://tc39.es/process-document/](https://tc39.es/process-document/)   
  * Examples  
     * Decorators Repo: [https://github.com/tc39/proposal-decorators](https://github.com/tc39/proposal-decorators)   

>      * Temporal Repo: [https://github.com/tc39/proposal-temporal](https://github.com/tc39/proposal-temporal)   

Reviewer: Why the extra space for this third-tier indent, here and below? Note that prior levels (and the "original" being changed here) are using 2 spaces per tier, a la --
```
* top
  * second
    * third
```

# Stage 2: Formalization

The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal draft.

  * What's needed from champions?
    * [ ] An [explainer](https://tag.w3.org/explainers/).
    * [ ] Documentation of alternatives and trade-offs considered.
    * [ ] Optionally, a draft specification text (or detailed examples/code samples if needed for clarity).
    * [ ] Early implementation experience, including prototypes or trials.
    * [ ] Evidence of stakeholder (e.g. web developers) demand and use-case alignment.

  * What's asked of the **Working Group**?
    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.

>     * [ ] A completed Working Draft for further iteration.

Reviewer 4: Link to `Working Draft` seemed useful, why was it removed?

Author: Ooops.

>     * [ ] A completed Working Draft for further iteration.

Reviewer 4: This seems at odds with the above `Optionally, a draft specification text (or detailed examples/code samples if needed for clarity).`. A Working Draft is a draft specification with minimal adoption from the WG, no? So how is the draft spec optional if the WD is required?

Author: You're right. Reverting that change.

>     * [ ] A completed Working Draft for further iteration.

Reviewer 4: Doesnt this mean that the spec PR is not optional as suggested above? How can you have "A completed Working Draft for further iteration" without "a draft specification text"

Reviewer 4: Or maybe Im just reading this wrong and the first is about prerequisites while the second is about exit criteria? I think "What's needed from champions?" and "What's asked of the **Working Group**?" is vague and should be made clearer about whether this is about prerequisites or exit criteria

    * [ ] Approval to dedicate the WG's time to Review spec PRs to merge the proposal into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).

  
# Stage 3: Implementation

> NOTE: We don't yet have enough experience as we get towards Stage 3 and 4, so they aren't fully baked yet. It is possible that these
> will change a lot as we learn, including potentially entirely replacing them with references to the W3C Process. 

The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR). The spec should be considered finished, pending editorial nit review. However, since multiple implementations are expected to be produced during this stage, it is possible there will be further changes to normative content based on feedback from those implementors.
 
  * What's needed from champions?
    * [ ] Proposal fully merged into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).
    * [ ] Web Platform Tests are available.

>     * [ ] At least two independent implementers and no unresolved objections (where resolution may include an agreement not to address the issue).

Reviewer 4: This requirement seems a bit off compared with the previous text `since multiple implementations are expected to be produced during this stage`. So are multiple implementors needed to reach stage 3 or could they possibly occur during stage 3? Note that in WHATWG it is two implementers supporting which is different from having actually implemented. https://whatwg.org/stages#stage3

Author: Agreed. Stage 3 is ready for implementation; so rephrasing to supporting makes sense.

Reviewer 4: Your agreement does not seem reflected in the text

     * \<model\> Repo: [https://github.com/immersive-web/model-element](https://github.com/immersive-web/model-element)  
     * Depth sensing Repo: [https://github.com/immersive-web/depth-sensing](https://github.com/immersive-web/depth-sensing)   
* WebAssembly  

>    * Proposals Repo: [https://github.com/WebAssembly/proposals](https://github.com/WebAssembly/proposals)  

Reviewer: ```suggestion
  * Proposals Repo: [https://github.com/WebAssembly/proposals](https://github.com/WebAssembly/proposals)  
```
